ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.js

```diff
@@ -21,13 +21,15 @@ class Menu extends React.Component {
   constructor(props) {
     super(props)
     const children = React.Children.toArray(props.children)
-    this.state = { active: this.chooseActive(children) }
+    this.state = { active: this.chooseActive(children, props.value) }
     this.onKeyDown = this.onKeyDown.bind(this)
   }
-  chooseActive(children, active) {
+  chooseActive(children, selection, active) {
```

Comment:
My main suggestion is that we move to passing in parameters in an object so order doesn't matter.
```
chooseActive({children, selection, active})
```

Although now that I think about it, I think chooseActive could work without passing any parameters.
```
chooseActive() {
  if (this.props.children.some(child => child.props.value === this.state.active)) {
    return this.state.active
  } else if (this.props.value) {
    return this.props.value
  } else if (this.props.children.length > 0) {
    return this.props.children[0].props.value
  } else {
    return undefined
  }
}
```

Reply:
I see some problems with this.
* When `chooseActive` is called in the constructor, `this.state` is not yet defined, so `this.state.active` will fail.
* When `chooseActive` is called in the constructor, `this.props.children` is passed through `React.Children.toArray`. I'm not sure if that call is necessary, though....

Also, is it safe to assume that `super(props)` populated `this.props`? I figured that this is the case, but if not, there are more propblems.

Reply:
After some discussion, we figured out solutions. `this.props` is set by `super(props)`, checking `this.state` before retrieving `this.state.active` works out, and it looks like the call to `toArray` was not necessary.